src/commands/schema/pull.js

```diff
@@ -0,0 +1,85 @@
+const FaunaCommand = require("../../lib/fauna-command.js");
+const fetch = require("node-fetch");
+const fs = require("fs");
+const path = require("path");
+const { Flags } = require("@oclif/core");
+
+class PullSchemaCommand extends FaunaCommand {
+  async run() {
+    const {
+      connectionOptions: { domain, port, scheme, secret },
+    } = await this.getClient();
+
+    const dir = this.flags.dir;
+
+    try {
+      const filesres = await fetch(
+        `${scheme}://${domain}:${port}/schema/1/files`,
+        {
+          method: "GET",
+          headers: { AUTHORIZATION: `Bearer ${secret}` },
+        }
+      );
+      const filesjson = await filesres.json();
+      if (filesjson.error) {
+        this.error(filesjson.error.message);
+      }
+      // Sort for consistent order. It's nice for tests.
+      const filenames = filesjson.files
+        .map((file) => file.filename)
+        .filter((name) => name.endsWith(".fsl"))
+        .sort();
+
+      // All of the below could be parallelized if necessary.
+      if (this.flags.delete) {
+        // Delete all .fsl files (not directories).
+        for (const deleteme of fs.readdirSync(dir)) {
+          const fp = path.join(dir, deleteme);
+          if (deleteme.endsWith(".fsl") && !fs.statSync(fp).isDirectory()) {
+            fs.unlinkSync(fp);
```

Comment:
lol this is a terrible function name, thank you javascript.

will this explode if you can't write the file? we might want some error handling to show the path and such.

also note that `statSync` can fail if you can't read the file, and I think that needs to be handled as well

Reply:
This is all inside the try, which will spit out the error on the console. Is that not sufficient?

Reply:
thats probably alright, but the error won't show the path of the file. It'd be nice to wrap this `if` in a try that shows logs `fp` on an error.

Reply:
It looks like it does show good information. Here's what it looked like when I tested push on an unknown directory:

```
$ bin/run schema push --dir=$HOME/aslkdfjgalksdfj
    Error: ENOENT: no such file or directory, scandir '/Users/wdberkeley/aslkdfjgalksdfj'
    Code: ENOENT
```